### PR TITLE
Issue/6586 document in not in operators

### LINKED ISSUE #6586: add documentation for "in" and "not in" operators
These operators are not documented in the language reference.

### PR DESCRIPTION
**Pull request by FloLey**
# Description

document 'in' and 'not in' operators

closes #6586 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: changelogs/unreleased/6586-document-in-and-not-in.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+description: Document the proper way to configure a project to access a private python package repository
+issue-nr: 6586
+change-type: patch
+destination-branches: [master]
```
Comment (FloLey) [inline changelogs/unreleased/6586-document-in-and-not-in.yml:4]: The "not in" operator is new right? it's not present in previous versions I mean?

File: changelogs/unreleased/6586-document-in-and-not-in.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+description: Add documentation for 'in' and 'not in' operators
+issue-nr: 6586
+change-type: patch
+destination-branches: [master]
```
Comment (sanderr) [inline changelogs/unreleased/6586-document-in-and-not-in.yml:4]: > The "not in" operator is new right? it's not present in previous versions I mean?

It is available on iso6

```suggestion
destination-branches: [master, iso6]
```

File: docs/language.rst [hunk 3]
```diff
@@ -195,10 +195,28 @@ value. It can have the following forms
         | value
         | value ('>' | '>=' | '<' | '<=' | '==' | '!=') value
         | value 'in' value
+        | value 'not in' value
```
Comment (sanderr) [inline docs/language.rst:198]:
```suggestion
        | value 'not' 'in' value
```

File: docs/language.rst [hunk 4]
```diff
@@ -195,10 +195,28 @@ value. It can have the following forms
...
+    myfiles = ["/a/b/c", "/c/d/e", "x/y/z/u/v/w"]
+
+    condition1 = "/a/b/c" in myfiles # evaluates to True
+    condition2 = "/f/g/h" in myfiles # evaluates to False
+
+    condition3 = "/a/b/c" not in myfiles # evaluates to False
+    condition4 = "/f/g/h" not in myfiles # evaluates to True
+
+    condition5 = not "/a/b/c" in myfiles # evaluates to False
+    condition6 = not "/f/g/h" in myfiles # evaluates to True
```
Comment (sanderr) [inline docs/language.rst:217]:
```suggestion
    condition1 = "/a/b/c" in myfiles # evaluates to true
    condition2 = "/f/g/h" in myfiles # evaluates to false

    condition3 = "/a/b/c" not in myfiles # evaluates to false
    condition4 = "/f/g/h" not in myfiles # evaluates to true

    condition5 = not ("/a/b/c" in myfiles) # evaluates to false
    condition6 = not ("/f/g/h" in myfiles) # evaluates to true
```
The parentheses aren't strictly required but I think it helps readability here to be explicit.

**Review by sanderr — APPROVED** [review]

**Pull request closed without merging**